index.ts

```diff
@@ -385,4 +382,67 @@ export enum WorkoutActivityType {
   Other = 3000,
 }
 
+export enum WorkoutMetadataKey {
+  ActivityType = "HKActivityType",
+  AppleFitnessPlusSession = "HKAppleFitnessPlusSession",
+  CoachedWorkout = "HKCoachedWorkout",
+  GroupFitness = "HKGroupFitness",
+  IndoorWorkout = "HKIndoorWorkout",
+  WorkoutBrandName = "HKWorkoutBrandName",
+  CyclingFunctionalThresholdPowerTestType = "HKCyclingFunctionalThresholdPowerTestType",
+  FitnessMachineDuration = "HKFitnessMachineDuration",
+  CrossTrainerDistance = "HKCrossTrainerDistance",
+  IndoorBikeDistance = "HKIndoorBikeDistance",
+  AverageMETs = "HKAverageMETs",
+  PhysicalEffortEstimationType = "HKPhysicalEffortEstimationType",
+  AlpineSlopeGrade = "HKAlpineSlopeGrade",
+  ElevationAscended = "HKElevationAscended",
+  ElevationDescended = "HKElevationDescended",
+  AverageSpeed = "HKAverageSpeed",
+  MaximumSpeed = "HKMaximumSpeed",
+  SwimmingLocationType = "HKSwimmingLocationType",
+  SwimmingStrokeStyle = "HKSwimmingStrokeStyle",
+  LapLength = "HKLapLength",
+  SWOLFScore = "HKSWOLFScore",
+  WaterSalinity = "HKWaterSalinity",
+}
+
+export type QuantityType = {
+  unit: HealthUnit | string;
+  doubleValue: number;
+}
+
+export enum WaterSalinityType {
+  freshWater = 0,
+  saltWater = 1,
+}
+
+export type WorkoutMetadata = {
+    // [WorkoutMetadataKey.ActivityType]?: string; // ?
```

Comment:
I left some of the keys commented due to missing documentation from Apple. Most of them are related to iOS 17, and the documentation is on beta.

Reply:
That's fine. Are these types not available yet?